pdns/ixfrdist-stats.hh

```diff
@@ -70,6 +71,11 @@ class ixfrdistStats {
       progStats.unknownDomainInQueries += 1;
     }
 
+    void incrementNotImplemented(const string& opcode)
+    {
+      notimpStats[opcode] ++;
```

Comment:
It looks like the map can be accessed by several threads at at time, which would be an issue because one thread can be inserting a new entry while another one is reading the map.

Reply:
good catch! making the values atomic is not enough

Reply:
given opcode is only 4 bits, an array might be simplest here?

Reply:
A `std::array` would work, sure! Given that we are not supposed to increment this value very often a mutex might work as well.

Reply:
done